.github/workflows/build.yml

```diff
@@ -55,6 +55,32 @@ jobs:
         run: gradle test
         working-directory: ./java
 
+      # Upload fs-storage JNI libs
+      - name: Install cargo-ndk
+        run: cargo install cargo-ndk
```

Comment:
I suggest creating a separate `android` job instead of including these steps in the `linux` job. This separation would reduce execution time and simplify the workflow structure. 

The android-specific build could run in parallel with other platform builds.

Reply:
That's what I did on the other Pull Request ([previous Pull Request](https://github.com/ARK-Builders/ark-core/pull/92#discussion_r1822047234)), but @kirillt sugestted to move all the setup to this one - `build.yml`, instead. While I still think moving it to a separate job is preferable, I'm interested in hearing @kirillt  reasoning for keeping it here.

Reply:
That's what I kind of did on the other Pull Request ([previous Pull Request](https://github.com/ARK-Builders/ark-core/pull/92#discussion_r1822047234)). While I still think moving it to a separate job is preferable, I'm interested in hearing what @kirillt thinks about it.

Reply:
I think @kirillt means to create a job in the same workflow, something like this:
```yaml
jobs:
  android:
    - uses: actions/checkout@v4
    - name: Add Rust Toolchain for android
    - name: Install cargo-ndk
    - name: Install JDK
    - name: Set up Gradle
    - name: Java tests
    - name: Build fs-storage JNI libs
    - name: Upload fs-storage JNI libs
    - name: Publish Java release

  linux:
    steps:
      - uses: actions/checkout@v4
      - name: Install Rust
      - name: Check
      - name: Format
      - name: Build Debug
      - name: Run tests
      - name: Build Release

  windows:
    steps:
      - uses: actions/checkout@v4
      - name: Install Rust
      - name: Build Release
      - name: Run tests

  mac-intel:
    steps:
      - uses: actions/checkout@v4
      - name: Install Rust
      - name: Build Release
      - name: Run tests

```

Reply:
In the past PR I actually was meaning one job for all-Java-related things because we had Java-related things scattered by 2 files (workflows).

Indeed, `android` job sounds like a good idea, but:
- initially, I thought that we keep `java` folder here with the idea of platform-agnostic bindings, i.e. same Java bindings could be used on desktop
- if this is not the case, then we should move the bindings to `ark-android` repo
- but given that Java bindings is temporary, we don't want to do redundant work, we even can wire `ark-android` to this branch of `ark-core` without merging into `main` (if it's possible)
- final goal is to have clean separation of Rust core which is unversal and multi-platform, from `ark-android` or other platform-dependent repos
- is UniFFI is quick to implement, we can remove `java` folder

Reply:
So, Java job should be separate and isolated from Rust jobs. Android job should be considered temporary because Core is multi-platform.

Reply:
No. All current jobs also build and test Rust code on each specific platform, which is important. 

Linux and Android will not be the same because the Linux job includes Rust tasks like `cargo build`, `cargo test` etc., while the Android job includes `cargo-ndk` build and other Android-specific tasks.

Reply:
Yes, @Pushkarm029, but maybe the missing point is: the Java tests and release depends on the built Rust code.

The Java sub-project is just JNI bindings for `fs-storage` and, in the future, if not replaced by `uniffi-rs`, it will also contain bindings for other crates 😬 So building and testing the Rust code is basically a "must have" too. The only extra thing compared to the Linux job would be the `cargo-ndk` related code

Reply:
Oh, yeah. I forgot about that. Thanks for explaining it!
> The only extra thing compared to the Linux job would be the cargo-ndk related code.


Yes, I think to simplify everything, we have to accept this. So, we would have something like:
```yml
linux: build test
android: build test cargo-ndk
mac: build test
windows: build test
```

Reply:
I think so.